src/main/java/com/google/cloud/genomics/utils/Contig.java

```diff
@@ -99,7 +99,7 @@ public Contig apply(String contigString) {
   // the ShardUtils methods should be used to ensure that shards are shuffled all together before
   // being returned to clients.
   List<Contig> getShards(long numberOfBasesPerShard) {
-    double shardCount = Math.ceil(end - start) / numberOfBasesPerShard;
+    double shardCount = (end - start) / (double) numberOfBasesPerShard;
```

Comment:
Thanks Nicole, but I am not contributing code, and not sure why my post just disappeared though.  This is a friendly discussion of exploratory ideas, and the code I posted was mostly your code with a proof-of-concept of why `Math.ceil` might not be useless.  This is just Java-logic that commonly discussed everywhere among all programmers.